src/tools/gen-multidom-structured-mesh.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+
+# gen-multidom-structured-mesh.py
+# Write a simple multidomain structured blueprint mesh for testing.
+
+# This script requires a conduit installation configured with python3 and hdf5.
+# Make sure PYTHONPATH includes /path/to/conduit/install/python-modules
+
+try:
+  import conduit
+  import conduit.blueprint
+  import conduit.relay
+except ModuleNotFoundError as e:
+  print(f'{e}\nMake sure your PYTHONPATH includes /path/to/conduit/install/python-modules\nConduit must be configured with python and hdf5.')
+  exit(-1)
+
+import numpy as np
+
+def i_c(s):
+  '''Integer coordinates.'''
+  return list(map(int, s.split(',')))
+def f_c(s):
+  '''Floating point coordinates.'''
+  return list(map(float, s.split(',')))
+
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+ps = ArgumentParser(description='Write a blueprint multidomain unstructured mesh.',
+                    formatter_class=ArgumentDefaultsHelpFormatter)
+ps.add_argument('--useList', action='store_true', help='Put domains in a list instead of a map')
+ps.add_argument('-ml', type=f_c, default=(0.,0.), help='Mesh lower coordinates')
+ps.add_argument('-mu', type=f_c, default=(1.,1.), help='Mesh upper coordinates')
+ps.add_argument('-ms', type=i_c, default=(3,3), help='Logical size of mesh (cells)')
+ps.add_argument('-dc', type=i_c, default=(1,1), help='Domain counts in each index direction')
+ps.add_argument('-o', '--output', type=str, default='mdmesh', help='Output file base name')
+opts,unkn = ps.parse_known_args()
+print(opts, unkn)
+if(unkn):
+  print("Unrecognized arguments:", *unkn)
+  quit(1)
+
+dim = len(opts.dc)
+
+if dim not in (2,3)	\
+   or len(opts.ms) != dim	\
+   or len(opts.ml) != dim	\
+   or len(opts.mu) != dim:
+  raise RuntimeError('dc, ms, ml and mu options must have the same dimensions (2 or 3)')
+
+def scale_structured_domain(n, startCoord, endCoord):
+  '''This funciton scales and shift a blueprint structured domain after
```

Comment:
typo: should be "This function"

Reply:
good catch. personally, I think we could use more funcitons in the code.... 😄